contracts/src/v0.8/vrf/VRFV2PlusWrapper.sol

```diff
@@ -0,0 +1,521 @@
+
+  // s_fulfillmentFlatFeeLinkPPM is the flat fee in millionths of LINK that VRFCoordinatorV2
+  // charges.
+  uint32 private s_fulfillmentFlatFeeLinkPPM;
+
+  // s_fulfillmentFlatFeeLinkPPM is the flat fee in millionths of LINK that VRFCoordinatorV2
+  // charges.
+  uint32 private s_fulfillmentFlatFeeEthPPM;
+
+  // Other configuration
+
+  // s_wrapperGasOverhead reflects the gas overhead of the wrapper's fulfillRandomWords
+  // function. The cost for this gas is passed to the user.
+  uint32 private s_wrapperGasOverhead;
+
+  // s_coordinatorGasOverhead reflects the gas overhead of the coordinator's fulfillRandomWords
+  // function. The cost for this gas is billed to the subscription, and must therefor be included
+  // in the pricing for wrapped requests. This includes the gas costs of proof verification and
+  // payment calculation in the coordinator.
+  uint32 private s_coordinatorGasOverhead;
+
+  // s_wrapperPremiumPercentage is the premium ratio in percentage. For example, a value of 0
+  // indicates no premium. A value of 15 indicates a 15 percent premium.
+  uint8 private s_wrapperPremiumPercentage;
+
+  // s_keyHash is the key hash to use when requesting randomness. Fees are paid based on current gas
+  // fees, so this should be set to the highest gas lane on the network.
+  bytes32 s_keyHash;
+
+  // s_maxNumWords is the max number of words that can be requested in a single wrapped VRF request.
+  uint8 s_maxNumWords;
+
+  struct Callback {
+    address callbackAddress;
+    uint32 callbackGasLimit;
+    uint256 requestGasPrice;
+  }
+  mapping(uint256 => Callback) /* requestID */ /* callback */ public s_callbacks;
+
+  constructor(
+    address _link,
+    address _linkEthFeed,
+    address _coordinator
+  ) ConfirmedOwner(msg.sender) VRFConsumerBaseV2Plus(_coordinator, msg.sender) {
+    if (_link != address(0)) {
+      s_link = LinkTokenInterface(_link);
+    }
+    if (_linkEthFeed != address(0)) {
+      s_linkEthFeed = AggregatorV3Interface(_linkEthFeed);
+    }
+    COORDINATOR = ExtendedVRFCoordinatorV2PlusInterface(_coordinator);
+
+    // Create this wrapper's subscription and add itself as a consumer.
+    uint64 subId = ExtendedVRFCoordinatorV2PlusInterface(_coordinator).createSubscription();
+    SUBSCRIPTION_ID = subId;
+    ExtendedVRFCoordinatorV2PlusInterface(_coordinator).addConsumer(subId, address(this));
+  }
+
+  /**
+   * @notice set the link token to be used by this wrapper
+   * @param link address of the link token
+   */
+  function setLINK(address link) external onlyOwner {
+    // Disallow re-setting link token because the logic wouldn't really make sense
+    if (address(s_link) != address(0)) {
+      revert LinkAlreadySet();
+    }
+    s_link = LinkTokenInterface(link);
+  }
+
+  /**
+   * @notice set the link eth feed to be used by this wrapper
+   * @param linkEthFeed address of the link eth feed
+   */
+  function setLinkEthFeed(address linkEthFeed) external onlyOwner {
```

Comment:
Similar question for V2+ coordinator: should we have an all-or-nothing setter for both LINK and the feed? Doesn't have to be solved in this PR.

Reply:
Yes.

Reply:
Let me just go ahead with these changes.